src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -35,95 +33,1003 @@ const ActorInit En_Sw_InitVars = {
+                }
+            } else {
+                if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame) == 1) {
+                    this->unk_38A = 2;
+                }
+                sp2C = 32768.0f / this->skelAnime.endFrame;
+                sp2C *= this->skelAnime.curFrame;
+                sp2C = Math_SinS(sp2C) * this->unk_420;
+                func_80B0CCF4(this, &sp2C);
+                this->actor.shape.rot = this->actor.world.rot;
+            }
+        }
+    }
+}
+
+void func_80B0D878(EnSw* this, GlobalContext* globalCtx) {
+    Actor* temp_v0;
+    Vec3f pos;
+    Vec3f velAndAccel = { 0.0f, 0.5f, 0.0f };
+    f32 x;
+    f32 y;
+    f32 z;
+
+    if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame) == 1) {
+        func_80B0CEA8(this, globalCtx);
+    }
+
+    func_80B0CCF4(this, &this->unk_420);
+    this->actor.shape.rot = this->actor.world.rot;
+    if (this->unk_394 == 0) {
+        if (this->unk_392 == 0) {
+            Audio_PlaySoundGeneral(NA_SE_SY_KINSTA_MARK_APPEAR, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+            x = (this->unk_364.x * 10.0f);
+            y = (this->unk_364.y * 10.0f);
+            z = (this->unk_364.z * 10.0f);
+            temp_v0 = Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_SI,
+                                         this->actor.world.pos.x + x, this->actor.world.pos.y + y,
+                                         this->actor.world.pos.z + z, 0, 0, 0, this->actor.params);
+            if (temp_v0 != NULL) {
+                temp_v0->parent = NULL;
+            }
+            Actor_Kill(&this->actor);
+            return;
+        }
+    }
+    if (this->unk_392 == 0) {
+        if (DECR(this->unk_394) != 0) {
+            pos = this->actor.world.pos;
+            pos.y += 10.0f + ((Rand_ZeroOne() - 0.5f) * 6.0f);
+            pos.x += (Rand_ZeroOne() - 0.5f) * 32.0f;
+            pos.z += (Rand_ZeroOne() - 0.5f) * 32.0f;
+            EffectSsDeadDb_Spawn(globalCtx, &pos, &velAndAccel, &velAndAccel, 42, 0, 255, 255, 255, 255, 255, 0, 0, 1,
+                                 9, 1);
+        }
+    }
+}
+
+void func_80B0DB00(EnSw* this, GlobalContext* globalCtx) {
+    Actor_MoveForward(&this->actor);
+    this->actor.shape.rot.x += 0x1000;
+    this->actor.shape.rot.z += 0x1000;
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 20.0f, 0.0f, 5);
+    if (this->actor.bgCheckFlags & 1) {
+        if (!(0.0f <= this->actor.velocity.y)) {
+            if (this->actor.floorHeight <= -32000.0f || this->actor.floorHeight >= 32000.0f) {
+                Actor_Kill(&this->actor);
+                return;
+            }
+
+            this->actor.bgCheckFlags &= ~1;
+            if (this->unk_38A == 0) {
+                this->actionFunc = func_80B0DC7C;
+                this->unk_394 = 10;
+            } else {
+                this->actor.velocity.y = ((this->unk_38A--) * 8.0f) * 0.5f;
```

Comment:
does the decrement have to be in there like that?

Reply:
yup